software/database/src/main/java/org/apache/brooklyn/entity/database/postgresql/PostgreSqlSshDriver.java

```diff
@@ -328,9 +340,39 @@ private void initializeNewDatabase() {
                         " --command="+ createUserCommand),
                 sudoAsUser("postgres", getInstallDir() + "/bin/psql -p " + entity.getAttribute(PostgreSqlNode.POSTGRESQL_PORT) + 
                                 " --command="+ createDatabaseCommand),
+                createRolesAdditionalCommand,
                 callPgctl("stop", true))
                 .failOnNonZeroResultCode().execute();
     }
+
+    private String buildCreateRolesQuery() {
+        Map<String, Object> roles = entity.getConfig(PostgreSqlNode.ROLES);
+        String createRolesQuery = "\"";
+
+        for (Map.Entry role: roles.entrySet()) {
+            createRolesQuery = createRolesQuery.concat(String.format("CREATE ROLE %s", role.getKey()));
+            if (((Map) role.getValue()).containsKey("properties")) {
+                createRolesQuery = createRolesQuery.concat(String.format(" WITH %s; ", ((Map) role.getValue()).get("properties")));
+            } else {
+                createRolesQuery = createRolesQuery.concat(";");
+            }
+
+            if (((Map) role.getValue()).containsKey("privilages")) {
```

Comment:
Thanks @geomacy 
Do you think the roles defaults are fine to be in brooklyn as they look pretty standard, not specific?